src/task/methods/scape/script.py

```diff
@@ -45,6 +48,24 @@
 # load log pvals
 df_de = scape.io.load_slogpvals(par['de_train']).drop(columns=["id", "split"], axis=1, errors="ignore")
 
+def confirm_celltype(cell, sm_name=None):
+	global df_de
+	cells = None
+	if sm_name is None:
+		cells = df_de.index.get_level_values("cell_type").unique()
+	else:
+		cells = df_de[df_de.index.get_level_values('sm_name')==sm_name].index.get_level_values("cell_type").unique()
+
+	if cell in cells:
+		return cell
+	else:
+		print(f"Input cell type ({cell}) not found in the" + f"drug {sm_name}" if sm_name is not None else "" + " data.")
+		cell_ = np.random.choice(cells)
+		print(f"Randomly selecting a cell type from the data: {cell_}.")
+		return cell_
+
+par["cell"] = confirm_celltype(par["cell"])
```

Comment:
if the other suggestion is accepted

```suggestion
par["cell"] = confirm_celltype(df_de, par["cell"])